spec/teacher_spec.rb

```diff
@@ -0,0 +1,25 @@
+require_relative '../teacher'
+
+describe Teacher do
```

Comment:
- [ ] Please don't forget to test the `can_use_services?` and  `to_json` methods since the requirements demand us to test for all methods in all classes.

Reply:
> Hi Team,
> 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
> 
> ![](https://camo.githubusercontent.com/0a2fa1881c9aa83a89f48445611c5cc6c3c0574c2a424daba275e90cbd3143a2/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6e59597451644c7177776c54653363634b682f67697068792e676966) [ ![68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6e59597451644c7177776c54653363634b682f67697068792e676966](https://camo.githubusercontent.com/0a2fa1881c9aa83a89f48445611c5cc6c3c0574c2a424daba275e90cbd3143a2/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6e59597451644c7177776c54653363634b682f67697068792e676966) ](https://camo.githubusercontent.com/0a2fa1881c9aa83a89f48445611c5cc6c3c0574c2a424daba275e90cbd3143a2/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6e59597451644c7177776c54653363634b682f67697068792e676966) [ ](https://camo.githubusercontent.com/0a2fa1881c9aa83a89f48445611c5cc6c3c0574c2a424daba275e90cbd3143a2/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6e59597451644c7177776c54653363634b682f67697068792e676966)
> 
> ## Highlights
> * All written tests are passing ✔️
> * Correct GitHub workflow 👍
> * No Linter error 👍
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

![image](https://user-images.githubusercontent.com/82343891/188004621-141b3b5b-2214-4d31-9d89-7f22e7f22e48.png)
The project requirements, as shown in the picture above, were to "Add unit tests to your classes". It never required us to write tests for _every_ method inside _every_ class. When looking at this review, we added other tests that we considered necessary. Therefore, we are going to request another review.

Reply:
Hi @Daf98 , in each class, we have methods. These methods bring about the functionality of our app. Therefore you should be testing for these methods :laughing: 


Once more why'll you select the methods to test knowing that these methods are the reasons for your app to function and you are carrying out tests for the functionality of your app :angry: